### PR TITLE
added new yaml for FunctionalTests-CI-PR

### LINKED ISSUE #300: Add minimal checks for PRs submitted to main
We currently don't have any checks on PRs submitted to main, I think that at the very minimum we should add a step that ensures all the solutions compile (dotnet, JS, Python and Functional Tests). 
We can add additional checks in the future.

### PR DESCRIPTION
**Pull request by harish0404**
Fixes #300 
Add minimal checks for PRs submitted to main 

•	dotnetBotsBuild-CI.yml: 
              o This YAML builds and restore the packages for SkillFunctionalTests.sln
•	jsBotsBuildCI.yml
             o	This YAML run NPM install and pack the JavaScript Bots
•	pythonBotsBuild-CI.yml
             o	This YAML will get the requirement.txt files for python and install dependencies.
•	dotnetBotsUpdateDependencies-CI.yml
             o	This YAML builds and restore the packages for FunctionalTestsBots
•	functionalTests-CI.yml
            o	This YAML runs all the above Dot Net, python, JS pipelines in parallel

### CODE REVIEW

File: Bots/DotNet/Skills/CodeFirst/EchoSkillBot/EchoSkillBot.csproj [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.1.2" />
-    <PackageReference Include="Microsoft.Bot.Builder.Dialogs" Version="4.11.1" />
+    <PackageReference Include="Microsoft.Bot.Builder.Dialogs" Version="4.13.0-daily.20210308.220944.384fea4" />
```
Comment (mrivera-ms) [inline Bots/DotNet/Skills/CodeFirst/EchoSkillBot/EchoSkillBot.csproj:16]: Why hard coding this version?

**Review by gabog — CHANGES_REQUESTED** [review]
Fix comma in functionalTests-CI,yml file name
Follow capitalization conventions for files names

File: build/yaml/functionalTestPython.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+  steps:
+  - task: UsePythonVersion@0
+    displayName: Use Python version
+    inputs:
+      versionSpec: '3.7.x'
```
Comment (gabog) [inline build/yaml/functionalTestPython.yml:5]: Change by
```suggestion
      versionSpec: '3.8.x'
```

File: build/yaml/functionalTests-CI.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# This runs under BotBuilder-Js-CI-yaml. Replaces classic build BotBuilder-JS-master-CI-node12.
```
Comment (gabog) [inline build/yaml/functionalTests-CI.yml:1]: This comment doesn't really describe what this file does, provide proper description

File: build/yaml/functionalTestPython.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+  steps:
```
Comment (gabog) [inline build/yaml/functionalTestPython.yml:1]: Rename this file as pythonBotsBuild-CI.yml

File: build/yaml/functionalTestJs.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
+steps:
```
Comment (gabog) [inline build/yaml/functionalTestJs.yml:1]: Rename this file as jsBotsBuild-CI.yml

File: build/yaml/functionalTestsBotsBuild.yml [hunk 6]
```diff
@@ -0,0 +1,233 @@
+steps:
```
Comment (gabog) [inline build/yaml/functionalTestsBotsBuild.yml:1]: I am not clear on what this yml does, can you explain?

File: build/yaml/functionalTests-CI.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+variables:
+  BuildConfiguration: 'Debug'
+  BuildPlatform: 'Any CPU'
+
+jobs:
+    - job: skillFunctionalTestsBuild
+      steps:
+      - template: skillFunctionalTestsBuild.yml
+
+    - job: functionalTestJs
```
Comment (gabog) [inline build/yaml/functionalTests-CI.yml:12]: Rename job
```suggestion
    - job: Build JS bots
```

File: build/yaml/functionalTests-CI.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+      steps:
+      - template: functionalTestJs.yml
+
+    - job: functionalTestPython
```
Comment (gabog) [inline build/yaml/functionalTests-CI.yml:16]: Rename job
```suggestion
    - job: Build Python bots
```

File: build/yaml/functionalTests-CI.yml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+      steps:
+      - template: functionalTestPython.yml
+
+    - job: functionalTestsBotsBuild
```
Comment (gabog) [inline build/yaml/functionalTests-CI.yml:20]: Rename job
```suggestion
    - job: Build dotnet bots
```

File: build/yaml/functionalTestsBotsBuild.yml [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+# Note: Template ci-build-steps.yml is not supported in macOS because it calls VSBuild@1 in order to build the Windows-only ASP.NET Desktop assemblies.
```
Comment (gabog) [inline build/yaml/functionalTestsBotsBuild.yml:2]: This comment don't seem to apply, remove?

File: build/yaml/functionalTestsBotsBuild.yml [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+  displayName: 'Publish Artifact: build folder'
+  inputs:
+    PathtoPublish: build
+    ArtifactName: build
+
+- powershell: |
+   cd ..
+   ls -R
+  displayName: 'Dir workspace'
+  continueOnError: true
```
Comment (gabog) [inline build/yaml/functionalTestsBotsBuild.yml:232]: Question: Should we continue on error?

File: build/yaml/skillFunctionalTestsBuild.yml [hunk 12]
```diff
@@ -0,0 +1,40 @@
+steps:
```
Comment (gabog) [inline build/yaml/skillFunctionalTestsBuild.yml:1]: rename this file as dotnetBotsBuild-CI.yml

File: build/yaml/skillFunctionalTestsBuild.yml [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+# Note: Template ci-build-steps.yml is not supported in macOS because it calls VSBuild@1 in order to build the Windows-only ASP.NET Desktop assemblies.
```
Comment (gabog) [inline build/yaml/skillFunctionalTestsBuild.yml:2]: This comment doesn't really apply here, consider removing it

**Review by gabog — CHANGES_REQUESTED** [review]
@harish0404 , please review and address the comments and make sure the build passes.

File: build/yaml/functionalTestPython.yml (hunk 2 above)
Comment (anumoluharish) [reply build/yaml/functionalTestPython.yml]: updated to 3.8.X

File: build/yaml/functionalTests-CI.yml (hunk 3 above)
Comment (anumoluharish) [reply build/yaml/functionalTests-CI.yml]: updated with new comment

File: build/yaml/functionalTestPython.yml (hunk 4 above)
Comment (anumoluharish) [reply build/yaml/functionalTestPython.yml]: updated with new name pythonBotsBuild-CI.yml

File: build/yaml/functionalTestJs.yml (hunk 5 above)
Comment (anumoluharish) [reply build/yaml/functionalTestJs.yml]: updated name with  jsBotsBuild-CI.yml

File: build/yaml/functionalTestsBotsBuild.yml (hunk 6 above)
Comment (anumoluharish) [reply build/yaml/functionalTestsBotsBuild.yml]: updated to dotnetBotsUpdateDependencies-CI.yml

File: Bots/DotNet/Skills/CodeFirst/EchoSkillBot/EchoSkillBot.csproj (hunk 1 above)
Comment (gabog) [reply Bots/DotNet/Skills/CodeFirst/EchoSkillBot/EchoSkillBot.csproj]: This one is preexisting and I added it to use the bits that fix a bug in dialogmanager, this should go away when we update to 4.12 as part of #321

File: build/yaml/functionalTests-CI.yml (hunk 9 above)
Comment (anumoluharish) [reply build/yaml/functionalTests-CI.yml]: updated  to Build dotnet bots

File: build/yaml/functionalTests-CI.yml (hunk 7 above)
Comment (anumoluharish) [reply build/yaml/functionalTests-CI.yml]: updated

File: build/yaml/functionalTests-CI.yml (hunk 8 above)
Comment (anumoluharish) [reply build/yaml/functionalTests-CI.yml]: updated

File: build/yaml/functionalTestsBotsBuild.yml (hunk 10 above)
Comment (anumoluharish) [reply build/yaml/functionalTestsBotsBuild.yml]: removed the comment

File: build/yaml/functionalTestsBotsBuild.yml (hunk 11 above)
Comment (anumoluharish) [reply build/yaml/functionalTestsBotsBuild.yml]: yes we can make it to false to stop on errors

File: build/yaml/skillFunctionalTestsBuild.yml (hunk 12 above)
Comment (anumoluharish) [reply build/yaml/skillFunctionalTestsBuild.yml]: renamed to dotnetBotsBuild-CI.yml

File: build/yaml/skillFunctionalTestsBuild.yml (hunk 13 above)
Comment (anumoluharish) [reply build/yaml/skillFunctionalTestsBuild.yml]: removed

**Review by gabog — APPROVED** [review]
I see some issues in the build outputs but will create separate issues to track them in another PR

**Pull request merged**